Create the Payment when the PayTomorrow order is prepared

abhishekgupta5 opened the pull request:
**Issue:** https://linear.app/nebulab-retainers/issue/ABU-16/create-the-payment-when-the-paytomorrow-order-is-prepared

- [x] QAed in sandbox

abhishekgupta5 committed:
Added HandleOrderCreationService

HandleOrderCreationService handles both creation of order
and creation of the payment object

Currently we don't use PaymentSource hence marked
source_required? as false for SolidusPayTomorrow::PaymentMethod
This is needed to create a payment object without PaymentSource

abhishekgupta5 committed:
Fixed flaky test

1. Used line_item.description directly instead of line_item.product.description
2. create_order_service_spec.rb:26 is flaky hence fixed that. SKU should
   be selected dynamically instead of hardcoded - SKU-1, SKU-2

abhishekgupta5 commented:
Payment object in checkout state -
<img width="585" alt="image" src="https://user-images.githubusercontent.com/9041163/196182799-fc5450ee-c065-41ec-b6df-d223e84df391.png">

DanielePalombo commented on app/services/solidus_pay_tomorrow/base_service.rb:
```diff
@@ -0,0 +1,13 @@
+
+module SolidusPayTomorrow
+  class BaseService
+    def initialize(**_args) end
```
Considering you added BaseService, what do you think to let to Client::BaseService to inherit from this Object?

ref https://github.com/nebulab/solidus_pay_tomorrow/blob/301245cdf773310a08d8d373683b644f3a5da76c/app/services/solidus_pay_tomorrow/client/base_service.rb#L7-L24

abhishekgupta5 replied:
Yes, was thinking of doing it!
Will do before merging.

abhishekgupta5 replied:
Done in 272ffaf5f9a4cd678157d668abab6f88587d85a1

DanielePalombo commented on app/services/solidus_pay_tomorrow/handle_order_creation_service.rb:
```diff
@@ -0,0 +1,36 @@
+
+    private
+
+    def create_order
```
NIT: I think that create_order_on_pay_tomorrow will be a better name here

abhishekgupta5 replied:
Done in de3d582fc3ea85827161f935a1388d6ae6f5ade4

DanielePalombo reviewed (approved):
The PR works for me.
I just left some suggestions to improve the PR.

abhishekgupta5 committed:
Inherit Client::BaseService from BaseService

SolidusPayTomorrow::BaseService can act as a base class for
SolidusPayTomorrow::Client::BaseService

DanielePalombo reviewed (approved):
LGTM
